### ISSUE TITLE
Plugin crashed with "invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by taminachen** [mattermost/mattermost-plugin-calls#514]
Hi, we found plugin crashed this morning. Here is the logs:
 
```sh
"debug [2023-08-31 20:12:32.598 Z] panic: runtime error: invalid memory address or nil pointer dereference caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] [signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x8afce0] caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z]                                               caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] goroutine 94943 [running]:                    caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"error [2023-08-31 20:12:32.598 Z] Plugin failed to ServeHTTP, RPC call failed   caller=""plugin/client_rpc.go:423"" plugin_id=com.mattermost.calls error=""unexpected EOF"""
"debug [2023-08-31 20:12:32.598 Z] github.com/mattermost/mattermost/server/public/model.(*AppError).Error(0x0) caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] 	github.com/mattermost/mattermost/server/public@v0.0.0-20230613002302-62a3ee8adcb5/model/utils.go:279 +0x40 caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] main.(*Plugin).handleGetAllChannels(0xc0001b4540, {0x1186c98, 0xc000b874e0}, 0xc00095e900) caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] 	github.com/mattermost/mattermost-plugin-calls/server/api.go:142 +0x645 caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] main.(*Plugin).ServeHTTP(0xc0001b4540, 0xfc7420?, {0x1186c98, 0xc000b874e0}, 0xc00095e900) caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] 	github.com/mattermost/mattermost-plugin-calls/server/api.go:568 +0x398 caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] github.com/mattermost/mattermost/server/public/plugin.(*hooksRPCServer).ServeHTTP(0xc0006d88e0, 0xc0006d8e40, 0x1?) caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] 	github.com/mattermost/mattermost/server/public@v0.0.0-20230613002302-62a3ee8adcb5/plugin/client_rpc.go:453 +0x417 caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] reflect.Value.call({0xc0006f8e40?, 0xc000012978?, 0x3?}, {0xfd1423, 0x4}, {0xc000802ef8, 0x3, 0xc000418180?}) caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"error [2023-08-31 20:12:32.598 Z] Plugin failed to ServeHTTP, RPC call failed   caller=""plugin/client_rpc.go:423"" plugin_id=com.mattermost.calls error=""unexpected EOF"""
"debug [2023-08-31 20:12:32.598 Z] 	reflect/value.go:584 +0x8c5                  caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] reflect.Value.Call({0xc0006f8e40?, 0xc000012978?, 0x0?}, {0xc000802ef8?, 0xc000a9ba80?, 0xc000802f42?}) caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] 	reflect/value.go:368 +0xbc                   caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] net/rpc.(*service).call(0xc00040c080, 0xc000044f00?, 0x0?, 0xc00060e280, 0xc00014cf00, 0xc000802f48?, {0xdec840?, 0xc0006d8e40?, 0xc00048c000?}, {0xdfcb40, ...}, ...) caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] 	net/rpc/server.go:382 +0x226                 caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] created by net/rpc.(*Server).ServeCodec       caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
"debug [2023-08-31 20:12:32.598 Z] 	net/rpc/server.go:479 +0x3fe                 caller=""plugin/hclog_adapter.go:54"" plugin_id=com.mattermost.calls"
```
It seems like a typo.
https://github.com/mattermost/mattermost-plugin-calls/blob/5166bf5af1309b5ec04a6c27d5ceaf829051e879/server/api.go#L128-L132
since early returned above, `appErr` must be `nil`.
https://github.com/mattermost/mattermost-plugin-calls/blob/5166bf5af1309b5ec04a6c27d5ceaf829051e879/server/api.go#L139-L143

Can you confirm that? Thanks.

Calls: 0.18.1
Mattermost: 7.8.8

**Comment by streamer45** [comment]
Thanks @taminachen , I believe we sent a fix for this already but likely it was missed in the release. We are looking into it.

**Comment by cpoile** [comment]
Hi @taminachen, I missed adding that fix to the v0.18.1 build, but the newly released [v0.18.2](https://github.com/mattermost/mattermost-plugin-calls/releases/tag/v0.18.2) has it. You can install manually or use the marketplace. Thank you for the report!
